hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileSystemDeleteHandlerTest.java

```diff
@@ -0,0 +1,247 @@
+        mockStore = mock(ReadableFileStoreImpl.class);
+        subject = new FileSystemDeleteHandler();
+
+        writableFileState = writableFileStateWithOneKey();
+        given(writableStates.<EntityNum, File>get(FILES)).willReturn(writableFileState);
+        writableStore = new WritableFileStoreImpl(writableStates);
+        config = new FilesConfig(101L, 121L, 112L, 111L, 122L, 102L, 123L, 1000000L, 1024);
+        lenient().when(handleContext.getConfiguration()).thenReturn(configuration);
+        lenient().when(configuration.getConfigData(FilesConfig.class)).thenReturn(config);
+    }
+
+    @Test
+    void returnsExpectedRecordBuilderType() {
+        assertInstanceOf(DeleteFileRecordBuilder.class, subject.newRecordBuilder());
+    }
+
+    @Test
+    @DisplayName("File not found returns error")
+    void fileIdNotFound() throws PreCheckException {
+        // given:
+        mockPayerLookup();
+        given(mockStore.getFileMetadata(notNull())).willReturn(null);
+        final var context = new FakePreHandleContext(accountStore, newSystemDeleteTxn());
+        context.registerStore(ReadableFileStoreImpl.class, mockStore);
+
+        // when:
+        assertThrowsPreCheck(() -> subject.preHandle(context), INVALID_FILE_ID);
+    }
+
+    @Test
+    @DisplayName("File without keys returns error")
+    void noFileKeys() throws PreCheckException {
+        // given:
+        mockPayerLookup();
+        mockFileLookup(null, mockStore);
+        final var context = new FakePreHandleContext(accountStore, newSystemDeleteTxn());
+        context.registerStore(ReadableFileStoreImpl.class, mockStore);
+
+        // when:
+        assertThrowsPreCheck(() -> subject.preHandle(context), UNAUTHORIZED);
+    }
+
+    @Test
+    @DisplayName("Fails handle if file doesn't exist")
+    void fileDoesntExist() {
+        final var txn = newSystemDeleteTxn().systemDeleteOrThrow();
+
+        writableFileState = emptyWritableFileState();
+        given(writableStates.<EntityNum, File>get(FILES)).willReturn(writableFileState);
+        writableStore = new WritableFileStoreImpl(writableStates);
+
+        final var msg =
+                assertThrows(
+                        HandleException.class,
+                        () -> subject.handle(handleContext, txn, writableStore));
+        assertEquals(INVALID_FILE_ID, msg.getStatus());
+    }
+
+    @Test
+    @DisplayName("Fails handle if the file is not a system file")
+    void fileIsNotSystemFile() {
+        final var txn = newFileDeleteTxn().systemDeleteOrThrow();
+
+        final var existingFile = writableStore.get(fileEntityNum.longValue());
+        assertTrue(existingFile.isPresent());
+        assertFalse(existingFile.get().deleted());
+
+        final var msg =
+                assertThrows(
+                        HandleException.class,
+                        () -> subject.handle(handleContext, txn, writableStore));
+        assertEquals(INVALID_FILE_ID, msg.getStatus());
+    }
```

Comment:
Should also assert here that the file is still no deleted.

Reply:
if the file id is greater than 750 it doesn't matter. I don't check the rest.
I can create a separate test that validates for file deleted.

Reply:
line 149 is doing exactly it `assertFalse(existingFile.get().deleted());`

Reply:
sure I will add it

Reply:
done